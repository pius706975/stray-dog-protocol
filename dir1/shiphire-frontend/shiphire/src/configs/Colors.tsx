export const Color = {
    bgColor: "#f0f6fb",
    softSecBgPrimary: "#bcd3e6",
    softGreyBgPrimary: "#cbcdd1",
    primaryColor: "#216dab",
    softPrimaryColor: "#6499c4",
    secColor: "#9cd2ff",
    softSecColor: "#d7edff",
    darkTextColor: "#4d5963",
    lightTextColor: "#f0f6fb",
    bgSuccessColor: "#e9faec",
    boldSuccessColor: "#1fcc44",
    successColor: "#4cd669",
    bgErrorColor: "#fae9e7",
    boldErrorColor: "#cc1f12",
    errorColor: "#d64c41",
    bgInfoColor: "#d0dbf5",
    boldInfoColor: "#124ccc",
    infoColor: "#5982db",
    bgWarningColor: "#fbf6e8",
    boldWarningColor: "#d9a318",
    warningColor: "#e1b546",
    bgNeutralColor: "#dbdee0",
    neutralColor: "#828b92",
    primaryDisableColor: "#6499c4"
}