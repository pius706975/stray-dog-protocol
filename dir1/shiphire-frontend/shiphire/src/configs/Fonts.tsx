export const FontFamily = {
    bold: "Kanit-Bold",
    boldItalic: "Kanit-BoldItalic",
    medium: "Kanit-Medium",
    mediumItalic: "Kanit-MediumItalic",
    regular: "Kanit-Regular",
    semiBold: "Kanit-SemiBold",
    semiBoldItalic: "Kanit-SemiBoldItalic"
}

export const FontSize = {
    xs: 14,
    sm: 16,
    md: 18,
    lg: 20,
    xl: 22,
    xl1: 24,
    xl2: 26,
    xxl: 30,
    xxxl: 36
}