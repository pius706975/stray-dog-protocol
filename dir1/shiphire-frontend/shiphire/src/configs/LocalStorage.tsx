import AsyncStorage from '@react-native-async-storage/async-storage';

export const TOKEN = '@token';
export const ROLES = '@roles';
export const ADMINDATA = '@admindata';
export const USERDATA = '@userdata';
export const COMPANYDATA = '@companydata';
export const SHIPOWNERDOCSUBMITTED = '@shipownerdocsubmitted';
export const RFQFILEPATH = '@rfqfilepath';
export const PROPOSALOWNERFILEPATH = '@proposalfilepath';
export const CONTRACTFILEPATH = '@contractfilepath';
export const USERLANGUAGE = '@userlanguage';
export const FCMTOKEN = '@fcmtoken';

export const setDataToLocalStorage = async (key: string, data: Object) => {
    try {
        const jsonData = JSON.stringify(data);
        return await AsyncStorage.setItem(key, jsonData);
    } catch (error) {
        return error;
    }
};

export const getDataFromLocalStorage = async (key: string) => {
    try {
        const jsonValue = await AsyncStorage.getItem(key);
        let valueObj;
        if (jsonValue) valueObj = JSON.parse(jsonValue);
        return valueObj;
    } catch (error) {
        return error;
    }
};

export const removeDataToLocalStorage = async (key: string) => {
    try {
        return await AsyncStorage.removeItem(key);
    } catch (error) {
        return error;
    }
};
