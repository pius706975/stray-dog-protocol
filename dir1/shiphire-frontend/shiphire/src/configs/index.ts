export * from "./Fonts"
export * from "./Colors"
export * from "./LocalStorage"
export * from "./Svg"
export * from "./StatusText"
export * from "./firebase/GoogleSignIn"
export * from "./firebase/AppleSignIn"