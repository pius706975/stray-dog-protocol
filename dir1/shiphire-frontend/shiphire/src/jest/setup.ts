import { useSubmitContract } from './../hooks/useShipOwner';
import { useSelector } from 'react-redux';
// include this line for mocking react-native-gesture-handler
import 'react-native-gesture-handler/jestSetup';
export * from '@react-native-async-storage/async-storage/jest/async-storage-mock';
import DocumentPicker, {
    DocumentPickerResponse,
} from 'react-native-document-picker';

jest.mock('react-native-reanimated', () => {
    const Reanimated = require('react-native-reanimated/mock');

    Reanimated.default.call = () => {};

    return Reanimated;
});

jest.mock('react-native/Libraries/Animated/NativeAnimatedHelper');

jest.mock('@react-native-async-storage/async-storage', () =>
    require('@react-native-async-storage/async-storage/jest/async-storage-mock'),
);

jest.mock('react-native-image-crop-picker', () => {
    openPicker: jest.fn(() => Promise.resolve({ path: 'path' }));
});

jest.mock('react-native-document-picker', () => {
    picker: jest.fn(() => Promise.resolve({ path: 'path' }));
});

jest.mock('@react-navigation/native', () => ({
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: jest.fn(),
}));

Object.defineProperty(window, 'FormData', {
    writable: true,
    value: jest.fn().mockImplementation(() => ({
        append: jest.fn(),
    })),
});

const mockStatusCodes = {
    SIGN_IN_CANCELLED: 'mock_SIGN_IN_CANCELLED',
    IN_PROGRESS: 'mock_IN_PROGRESS',
    PLAY_SERVICES_NOT_AVAILABLE: 'mock_PLAY_SERVICES_NOT_AVAILABLE',
    SIGN_IN_REQUIRED: 'mock_SIGN_IN_REQUIRED',
};

jest.mock('@react-native-google-signin/google-signin', () => ({
    statusCodes: mockStatusCodes,
}));

jest.mock('@react-native-firebase/auth', () => ({
    GoogleAuthProvider: {
        credential: jest.fn().mockReturnValue('123'),
    },
}));

jest.mock('@notifee/react-native', () => {});
jest.mock('@react-native-firebase/messaging', () => {});

let mockDocumentPickerResolve = true;
const setMockDocumentPickerResolve = (value: boolean) => {
    mockDocumentPickerResolve = value;
};
jest.mock('react-native-document-picker', () => ({
    pick: jest.fn(() => {
        if (mockDocumentPickerResolve) {
            return Promise.resolve([{ uri: 'dummy.pdf', name: 'dummy.pdf' }]);
        } else {
            return Promise.reject(new Error('Error selecting PDF'));
        }
    }),
    types: {
        pdf: 'application/pdf',
    },
}));

let mockImagePickerResolve = true;
const setMockImagePickerResolve = (value: boolean) => {
    mockImagePickerResolve = value;
};
jest.mock('react-native-image-crop-picker', () => ({
    openPicker: jest.fn(() => {
        if (mockImagePickerResolve) {
            return Promise.resolve([{ uri: 'dummy.jpg', name: 'dummy.jpg' }]);
        } else {
            return Promise.reject(new Error('Error selecting Image'));
        }
    }),
    types: {
        pdf: 'application/pdf',
    },
}));

let mockImagePickerPathResolve = true;
const setMockImagePickerPathResolve = (value: boolean) => {
    mockImagePickerPathResolve = value;
};
jest.mock('react-native-image-crop-picker', () => ({
    openPicker: jest.fn(() => {
        if (mockImagePickerPathResolve) {
            return Promise.resolve([{ 
                creationDate: '1522437259',
                cropRect: null,
                data: null,
                duration: null,
                exif: null,
                filename: 'IMG_0006.HEIC',
                height: 3024,
                localIdentifier: 'CC95F08C-88C3-4012-9D6D-64A413D254B3/L0/001',
                mime: 'image/heic',
                modificationDate: '1530944178',
                path: '/Users/tabrezakhlaque/Library/Developer/CoreSimulator/Devices/3A21CC65-FCBF-4B7B-BE2C-87E18F17D6DC/data/Containers/Data/Application/C34290DF-2011-480C-9076-2B3CCFA8CD17/tmp/react-native-image-crop-picker/B02EB928-5871-4616-BA8F-05EF93037EB1.jpg',
                size: 2808983,
                sourceURL:
                    'file:///Users/tabrezakhlaque/Library/Developer/CoreSimulator/Devices/3A21CC65-FCBF-4B7B-BE2C-87E18F17D6DC/data/Media/DCIM/100APPLE/IMG_0006.HEIC',
                width: 4032,
             }]);
        } else {
            return Promise.reject(new Error('Error selecting Image'));
        }
    }),
    types: {
        image: 'image/jpg',
    },
}));

jest.mock('@gorhom/bottom-sheet', () => ({
    __esModule: true,
    ...require('@gorhom/bottom-sheet/mock'),
}));
jest.mock('react-i18next', () => ({
    useTranslation: () => ({
        t: key => key,
    }),
}));

jest.useFakeTimers();

export { setMockDocumentPickerResolve, setMockImagePickerResolve, setMockImagePickerPathResolve };
