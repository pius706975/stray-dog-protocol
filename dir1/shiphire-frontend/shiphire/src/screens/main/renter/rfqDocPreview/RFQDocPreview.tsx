import mime from 'mime';
import React from 'react';
import { useTranslation } from 'react-i18next';
import { Platform } from 'react-native';
import { View } from 'react-native-ui-lib';
import PDFView from 'react-native-view-pdf';
import { useDispatch } from 'react-redux';
import { Button, ScreenLayout } from '../../../../components';
import {
    Color,
    RFQFILEPATH,
    SendIcon,
    getDataFromLocalStorage,
} from '../../../../configs';
import { useSubmitRequestForQuote } from '../../../../hooks';
import { modalSlice, progressIndicatorSlice } from '../../../../slices';
import { RFQDocPreviewProps } from '../../../../types';
import { handleAxiosError } from '../../../../utils';

const RFQDocPreview: React.FC<RFQDocPreviewProps> = ({ navigation, route }) => {
    const dispatch = useDispatch();
    const {
        shipId,
        categoryId,
        shipOwnerId,
        rentalDuration,
        rentalDate,
        needs,
        locationDeparture,
        locationDestination,
        shipRentType,
    } = route.params;
    const mutationSubmitRFQ = useSubmitRequestForQuote();
    const { hideModal, showModal } = modalSlice.actions;
    const [pdfFilePath, setPdfFilePath] = React.useState<string>('');
    const [pdfFilePathiOS, setPdfFilePathiOS] = React.useState<string>('');
    const formData = new FormData();
    const { showProgressIndicator, hideProgressIndicator } =
        progressIndicatorSlice.actions;

    const { t } = useTranslation('rfq');
    const newPdfFilePath =
        Platform.OS === 'ios' ? pdfFilePathiOS : 'file://' + pdfFilePath;

    formData.append('document', {
        uri: newPdfFilePath,
        type: mime.getType(newPdfFilePath),
        name: `RFQ-${shipId}-${shipId}`,
    });

    formData.append('shipId', shipId);
    formData.append('categoryId', categoryId);
    formData.append('shipOwnerId', shipOwnerId);
    formData.append('rentalDuration', rentalDuration);
    formData.append('rentalDate', rentalDate);
    formData.append('needs', needs);
    formData.append('locationDeparture', locationDeparture);
    formData.append('locationDestination', locationDestination);
    formData.append('shipRentType', shipRentType);

    const handleSendRFQ = () => {
        dispatch(showProgressIndicator());
        mutationSubmitRFQ.mutate(formData, {
            onSuccess: () => {
                dispatch(
                    showModal({
                        status: 'success',
                        text: t('RFQDocPreview.successRFQSent'),
                    }),
                );
                setTimeout(() => {
                    navigation.reset({
                        index: 0,
                        routes: [{ name: 'MainScreenTab' }],
                    });
                    dispatch(hideModal());
                    dispatch(hideProgressIndicator());
                }, 3000);
            },
            onError: err => {
                dispatch(
                    showModal({
                        status: 'failed',
                        text: t('RFQDocPreview.failedSendRFQ'),
                    }),
                );
                setTimeout(() => {
                    dispatch(hideModal());
                    dispatch(hideProgressIndicator());
                }, 2000);
                handleAxiosError(err);
            },
        });
    };

    React.useEffect(() => {
        async function fetchPdfFilePath() {
            try {
                const resp = await getDataFromLocalStorage(RFQFILEPATH);
                if (resp && resp.path) {
                    if (Platform.OS === 'ios') {
                        const pdfPath = resp.path.split('/');
                        const fileName = pdfPath[pdfPath.length - 1];
                        setPdfFilePathiOS(resp.path);
                        setPdfFilePath(fileName);
                    } else {
                        setPdfFilePath(resp.path);
                    }
                } else {
                    console.error('Failed to retrieve PDF file path.');
                }
            } catch (error) {
                console.error('Error fetching PDF file path:', error);
            }
        }

        fetchPdfFilePath();
    }, []);

    return (
        <ScreenLayout
            testId="RFQDocPreviewScreen"
            backgroundColor="light"
            padding={10}>
            {pdfFilePath && (
                <View
                    style={{
                        flex: 1,
                        backgroundColor: Color.bgColor,
                    }}>
                    <PDFView
                        style={{
                            flex: 1,
                            borderWidth: 1,
                            backgroundColor: Color.bgColor,
                        }}
                        fadeInDuration={250.0}
                        resource={pdfFilePath}
                        {...(Platform.OS === 'ios' && {
                            fileFrom: 'documentsDirectory',
                        })}
                        resourceType={'file'}
                        // onLoad={() => console.log(`PDF rendered from ${pdfFilePath}`)}
                        onError={error =>
                            console.log('Cannot render PDF', error)
                        }
                    />
                </View>
            )}
            <View
                style={{
                    position: 'absolute',
                    bottom: 30,
                    right: 20,
                }}>
                <Button
                    testID="SendButton"
                    onSubmit={handleSendRFQ}
                    title={t('RFQDocPreview.labelButtonRFQ')}
                    color="success"
                    leftIcon={<SendIcon />}
                    isSubmitting={mutationSubmitRFQ.isLoading}
                />
            </View>
        </ScreenLayout>
    );
};

export default RFQDocPreview;
