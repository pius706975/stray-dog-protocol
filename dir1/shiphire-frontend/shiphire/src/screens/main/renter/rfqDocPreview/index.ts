export { default as RFQDocPreview } from "./RFQDocPreview"