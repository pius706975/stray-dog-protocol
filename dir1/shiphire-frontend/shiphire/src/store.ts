import { configureStore } from '@reduxjs/toolkit';
import {
    addShipSlice,
    documentNameSlice,
    modalSlice,
    networkErrorSlice,
    notifBadgeSlice,
    progressIndicatorSlice,
    renterPreferenceSlice,
    requestForQuoteSlice,
    shipSpecSlice,
    termsStatusSlice,
    userLanguageSlice,
    userLocationSlice,
    userStatusSlice,
} from './slices';

const store = configureStore({
    reducer: {
        modal: modalSlice.reducer,
        progressIndicator: progressIndicatorSlice.reducer,
        renterPreference: renterPreferenceSlice.reducer,
        requestForQuote: requestForQuoteSlice.reducer,
        userStatus: userStatusSlice.reducer,
        addShip: addShipSlice.reducer,
        documentName: documentNameSlice.reducer,
        termsStatus: termsStatusSlice.reducer,
        userLanguage: userLanguageSlice.reducer,
        userLocation: userLocationSlice.reducer,
        shipSpecSlice: shipSpecSlice.reducer,
        notifBadgeSlice: notifBadgeSlice.reducer,
        networkError: networkErrorSlice.reducer,
    },
});

export default store;
