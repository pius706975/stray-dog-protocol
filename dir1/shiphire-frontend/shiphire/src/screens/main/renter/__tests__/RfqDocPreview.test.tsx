import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { QueryClient, QueryClientProvider } from 'react-query';
import {
    MainStackParamList,
    RequestForAQuoteParamList,
} from '../../../../types';
import MockAdapter from 'axios-mock-adapter';
import httpRequest from '../../../../services/api';
import { Provider } from 'react-redux';
import store from '../../../../store';
import { NavigationContainer } from '@react-navigation/native';
import { RFQDocPreview } from '../rfqDocPreview';
import { fireEvent, render, waitFor } from '@testing-library/react-native';
import AsyncStorage from '@react-native-async-storage/async-storage';
import { RFQFILEPATH, USERDATA } from '../../../../configs';
import { setMockDocumentPickerResolve } from '../../../../jest/setup';

const queryClient = new QueryClient();
const MainStack = createNativeStackNavigator<RequestForAQuoteParamList>();
const mockAdapter = new MockAdapter(httpRequest);

const rfqMockData = {
    shipId: '1',
    categoryId: 'mock category id',
    shipOwnerId: 'mock ship owner id',
    rentalDuration: 20,
    rentalDate: 'mock rental date',
    needs: 'mock needs',
    locationDeparture: 'mock location departure',
    locationDestination: 'mock location destination',
    shipRentType: 'mock ship rent type',
};

const rfqMockScreen = (
    <QueryClientProvider client={queryClient}>
        <Provider store={store}>
            <NavigationContainer>
                <MainStack.Navigator>
                    <MainStack.Screen
                        name="DocPreview"
                        component={RFQDocPreview}
                        initialParams={rfqMockData}
                    />
                </MainStack.Navigator>
            </NavigationContainer>
        </Provider>
    </QueryClientProvider>
);
AsyncStorage.setItem(
    RFQFILEPATH,
    JSON.stringify({
        path: '/storage/emulated/0/Android/data/id.smiteknologi.shiphire/files/Documents/RFQ-Azis-undefined.pdf',
    }),
);

describe('testing RFQDocPreview', () => {
    describe('snapshot testing', () => {
        it('should render correctly', () => {
            const tree = render(rfqMockScreen).toJSON();
            expect(tree).toMatchSnapshot();
        });
    });
    describe('render testing', () => {
        it('should render correctly', () => {
            render(rfqMockScreen);

            expect('RFQDocPreview').toBeTruthy();
        });
        it('should submit after clicking send button', () => {
            const { getByTestId } = render(rfqMockScreen);

            const sendButton = getByTestId('SendButton');
            fireEvent.press(sendButton);
        });
        // it.only('should submit after clicking send button and show the modal', async () => {
        //     const { getByTestId } = render(rfqMockScreen);

        //     const sendButton = getByTestId('SendButton');
        //     fireEvent.press(sendButton);

        //     mockAdapter
        //         .onPost('/renter/submit-request-for-quote')
        //         .reply(200, { status: 'success', data: rfqMockData });

        //     const progressState = store.getState().progressIndicator;
        //     expect(progressState).toEqual({
        //         visible: true,
        //     });
        //     const modalState = store.getState().modal;

        //     await waitFor(() => {
        //         jest.advanceTimersByTime(3000);
        //         expect(modalState).toEqual({
        //             visible: true,
        //             status: 'success',
        //             text: 'RFQDocPreview.successRFQSent',
        //         });
        //         const progressState = store.getState().progressIndicator;
        //         expect(progressState).toEqual({
        //             visible: false,
        //         });
        //     });
        // });
        it('should failed submit after clicking send button and show the modal', async () => {
            const { getByTestId } = render(rfqMockScreen);

            const sendButton = getByTestId('SendButton');
            fireEvent.press(sendButton);

            mockAdapter
                .onPost('/renter/submit-request-for-quote')
                .reply(200, { status: 'failed' });

            const progressState = store.getState().progressIndicator;
            expect(progressState).toEqual({
                visible: true,
            });
            const modalState = store.getState().modal;

            await waitFor(() => {
                jest.advanceTimersByTime(3000);
                expect(modalState).toEqual({
                    visible: true,
                    status: 'failed',
                    text: 'RFQDocPreview.failedSendRFQ',
                });
                const progressState = store.getState().progressIndicator;
                expect(progressState).toEqual({
                    visible: false,
                });
            });
        });
    });
});
