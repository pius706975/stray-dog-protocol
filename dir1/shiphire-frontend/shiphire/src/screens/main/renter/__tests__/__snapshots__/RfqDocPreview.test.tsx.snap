// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing RFQDocPreview snapshot testing should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    {
      "flex": 1,
    }
  }
>
  <RNSScreenStack
    style={
      {
        "flex": 1,
      }
    }
  >
    <RNSScreen
      collapsable={false}
      forwardedRef={[Function]}
      gestureResponseDistance={
        {
          "bottom": -1,
          "end": -1,
          "start": -1,
          "top": -1,
        }
      }
      nativeBackButtonDismissalEnabled={false}
      onAppear={[Function]}
      onDisappear={[Function]}
      onDismissed={[Function]}
      onGestureCancel={[Function]}
      onHeaderBackButtonClicked={[Function]}
      onNativeDismissCancelled={[Function]}
      onTransitionProgress={[Function]}
      onWillDisappear={[Function]}
      replaceAnimation="push"
      sheetAllowedDetents="large"
      sheetCornerRadius={-1}
      sheetExpandsWhenScrolledToEdge={true}
      sheetGrabberVisible={false}
      sheetLargestUndimmedDetent="all"
      stackPresentation="push"
      style={
        {
          "bottom": 0,
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
      swipeDirection="horizontal"
    >
      <View
        accessibilityElementsHidden={false}
        importantForAccessibility="auto"
        style={
          {
            "flex": 1,
            "flexDirection": "column-reverse",
          }
        }
      >
        <View
          collapsable={false}
          style={
            [
              {
                "flex": 1,
              },
              {
                "backgroundColor": "rgb(242, 242, 242)",
              },
              undefined,
            ]
          }
        >
          <View
            style={
              [
                undefined,
                undefined,
                {
                  "flex": 1,
                },
                undefined,
                {},
                {},
                {},
                {
                  "backgroundColor": "#f0f6fb",
                  "justifyContent": "space-between",
                  "position": "relative",
                },
              ]
            }
            testID="RFQDocPreviewScreen"
          >
            <RCTScrollView
              contentContainerStyle={
                {
                  "alignItems": undefined,
                  "flex": undefined,
                  "flexGrow": 1,
                  "gap": undefined,
                  "justifyContent": undefined,
                  "marginBottom": undefined,
                  "padding": 10,
                  "paddingVertical": undefined,
                }
              }
            >
              <View>
                <View
                  style={
                    [
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      undefined,
                      {
                        "bottom": 30,
                        "position": "absolute",
                        "right": 20,
                      },
                    ]
                  }
                >
                  <View
                    accessibilityState={
                      {
                        "busy": undefined,
                        "checked": undefined,
                        "disabled": undefined,
                        "expanded": undefined,
                        "selected": undefined,
                      }
                    }
                    accessibilityValue={
                      {
                        "max": undefined,
                        "min": undefined,
                        "now": undefined,
                        "text": undefined,
                      }
                    }
                    accessible={true}
                    collapsable={false}
                    focusable={true}
                    onBlur={[Function]}
                    onClick={[Function]}
                    onFocus={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      {
                        "alignItems": "center",
                        "backgroundColor": "#1fcc44",
                        "borderRadius": 6,
                        "elevation": 2,
                        "flexDirection": "row",
                        "gap": 10,
                        "justifyContent": "center",
                        "paddingHorizontal": 10,
                        "paddingVertical": 6,
                      }
                    }
                    testID="SendButton"
                  >
                    <RNSVGSvgView
                      align="xMidYMid"
                      bbHeight="20"
                      bbWidth="20"
                      fill="none"
                      focusable={false}
                      height={20}
                      meetOrSlice={0}
                      minX={0}
                      minY={0}
                      style={
                        [
                          {
                            "backgroundColor": "transparent",
                            "borderWidth": 0,
                          },
                          {
                            "flex": 0,
                            "height": 20,
                            "width": 20,
                          },
                        ]
                      }
                      vbHeight={20}
                      vbWidth={20}
                      width={20}
                      xml="<svg width="20" height="20" viewBox="0 0 20 20" fill="none" xmlns="http://www.w3.org/2000/svg">
    <g clip-path="url(#clip0_699_317)">
    <path d="M2.83317 17L17.3748 10.7667C18.0498 10.475 18.0498 9.52499 17.3748 9.23333L2.83317 2.99999C2.28317 2.75833 1.67484 3.16666 1.67484 3.75833L1.6665 7.59999C1.6665 8.01666 1.97484 8.37499 2.3915 8.42499L14.1665 9.99999L2.3915 11.5667C1.97484 11.625 1.6665 11.9833 1.6665 12.4L1.67484 16.2417C1.67484 16.8333 2.28317 17.2417 2.83317 17V17Z" fill="#f0f6fb"/>
    </g>
    <defs>
    <clipPath id="clip0_699_317">
    <rect width="20" height="20" fill="white"/>
    </clipPath>
    </defs>
    </svg>        
    "
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <RNSVGGroup
                        fill={null}
                        propList={
                          [
                            "fill",
                          ]
                        }
                      >
                        <RNSVGGroup
                          clipPath="clip0_699_317"
                          fill={
                            {
                              "payload": 4278190080,
                              "type": 0,
                            }
                          }
                        >
                          <RNSVGPath
                            d="M2.83317 17L17.3748 10.7667C18.0498 10.475 18.0498 9.52499 17.3748 9.23333L2.83317 2.99999C2.28317 2.75833 1.67484 3.16666 1.67484 3.75833L1.6665 7.59999C1.6665 8.01666 1.97484 8.37499 2.3915 8.42499L14.1665 9.99999L2.3915 11.5667C1.97484 11.625 1.6665 11.9833 1.6665 12.4L1.67484 16.2417C1.67484 16.8333 2.28317 17.2417 2.83317 17V17Z"
                            fill={
                              {
                                "payload": 4293981947,
                                "type": 0,
                              }
                            }
                            propList={
                              [
                                "fill",
                              ]
                            }
                          />
                        </RNSVGGroup>
                        <RNSVGDefs>
                          <RNSVGClipPath
                            fill={
                              {
                                "payload": 4278190080,
                                "type": 0,
                              }
                            }
                            name="clip0_699_317"
                          >
                            <RNSVGRect
                              fill={
                                {
                                  "payload": 4294967295,
                                  "type": 0,
                                }
                              }
                              height="20"
                              propList={
                                [
                                  "fill",
                                ]
                              }
                              width="20"
                              x="0"
                              y="0"
                            />
                          </RNSVGClipPath>
                        </RNSVGDefs>
                      </RNSVGGroup>
                    </RNSVGSvgView>
                    <Text
                      numberOfLines={0}
                      style={
                        {
                          "borderBottomWidth": 0,
                          "borderColor": undefined,
                          "color": "#f0f6fb",
                          "fontFamily": "Kanit-Medium",
                          "fontSize": 18,
                          "lineHeight": undefined,
                          "textAlign": undefined,
                        }
                      }
                    >
                      RFQDocPreview.labelButtonRFQ
                    </Text>
                    <View
                      style={
                        [
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          undefined,
                          {
                            "position": "absolute",
                            "right": 20,
                          },
                        ]
                      }
                    />
                  </View>
                </View>
              </View>
            </RCTScrollView>
          </View>
        </View>
      </View>
      <RNSScreenStackHeaderConfig
        backButtonInCustomView={false}
        backTitleVisible={true}
        backgroundColor="rgb(255, 255, 255)"
        color="rgb(0, 122, 255)"
        direction="ltr"
        disableBackButtonMenu={false}
        hidden={false}
        hideBackButton={false}
        largeTitleHideShadow={false}
        title="DocPreview"
        titleColor="rgb(28, 28, 30)"
        topInsetEnabled={false}
        translucent={false}
      />
    </RNSScreen>
  </RNSScreenStack>
</RNCSafeAreaProvider>
`;
